### DISCUSSION TITLE
Is it possible to build a generic binary with default dist resources and could change without re-build later?

### DISCUSSION THREAD

**Discussion by liudonghua123** [tauri-apps/tauri#8379] [General]
Hi, I write some tools in pure html/css/js, and I want to use tauri to package them as binary. But sometimes I only changed the dist resources, name and icon, I need to change tauri.config.json file and re-build it.

Is there any methods for just replace the dist resource and icon in the prebuild binary? In such way, the process I mentioned above is more easier and quick.

**Comment by FabianLars** [comment]
In dev mode you could use a tool like vite that has hot reloading but i assume you're talking about `tauri build`. In this case rebuilding is required to get these files into the binary.
Loading new/unknown files at runtime (without copying them into the binary) is tracked here https://github.com/tauri-apps/tauri/issues/5302, the alternative to that is to just have a small localhost server (preferably in rust, alternatively as a sidecar) that serves these files.

**Reply by liudonghua123** [reply]
well, is there any chance that I can update the asset resources in the binary without re-build from scratch?

I don't know how tauri package the asset resources in the binary, can I just separate this step and for new related app, I can only perform this process step.

**Reply by FabianLars** [reply]
No, the asset inlining is part of the rust compilation. Overwriting that part of the binary afterwards is just not possible.
